microgpt.py

```diff
@@ -156,58 +151,56 @@
         if command == "talk_to_user":
             print(colored(f"MicroGPT: {arg}", 'cyan'))
             user_input = input('Your response: ')
-            memory.add(f"{mem}The user responded with: {user_input}.")
+            agent.memorize(f"{mem}The user responded with: {user_input}.")
             continue
 
         _arg = arg.replace("\n", "\\n") if len(arg) < 64 else f"{arg[:64]}...".replace("\n", "\\n")
         print(colored(f"MicroGPT: {thought}\nCmd: {command}, Arg: \"{_arg}\"", "cyan"))
         user_input = input('Press enter to perform this action or abort by typing feedback: ')
 
         if len(user_input) > 0:
-            memory.add(f"{mem}The user responded: {user_input}."\
+            agent.memorize(f"{mem}The user responded: {user_input}."\
                 "Take this comment into consideration.")
             continue
         try:
             if command == "execute_python":
                 _stdout = StringIO()
                 with redirect_stdout(_stdout):
                     exec(arg)
-                memory.add(f"{mem}{_stdout.getvalue()}")
+                agent.memorize(f"{mem}{_stdout.getvalue()}")
             elif command == "execute_shell":
                 result = subprocess.run(arg, capture_output=True, shell=True, check=False)
-                memory.add(f"{mem}STDOUT:\n{result.stdout}\nSTDERR:\n{result.stderr}")
+                agent.memorize(f"{mem}STDOUT:\n{result.stdout}\nSTDERR:\n{result.stderr}")
             elif command == "web_search":
-                memory.add(f"{mem}{ddg(arg, max_results=5)}")
+                agent.memorize(f"{mem}{ddg(arg, max_results=5)}")
             elif command == "web_scrape":
                 with urlopen(arg) as response:
                     html = response.read()
 
-                response_text = memory.summarize_memory_if_large(
-                    BeautifulSoup(
+                response_text = agent.chunked_summarize(
+                    content=BeautifulSoup(
                         html,
                         features="lxml"
                     ).get_text(),
-                    max_memory_item_size,
-                    summarizer_hint=objective
+                    max_tokens=max_memory_item_size,
+                    instruction_hint=SUMMARY_HINT + EXTRA_SUMMARY_HINT.format(objective=objective)
```

Comment:
Gererates a KeyError:

```
Traceback (most recent call last):
  File "/Users/bernhardmueller/Projects/micro-gpt/microgpt.py", line 186, in <module>
    instruction_hint=SUMMARY_HINT + EXTRA_SUMMARY_HINT.format(objective=objective)
KeyError: 'summarizer_hint'
```

I assume this should be `summarizer_hint=objective`?